jenkins_rock/rockcraft.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: jenkins
+summary: Jenkins rock
+description: Jenkins OCI image for the Jenkins charm
+version: "1.0"
+base: ubuntu:22.04
+build-base: ubuntu:22.04
+license: Apache-2.0
+platforms:
+  amd64:
+services:
+  jenkins:
+    override: merge
+    summary: The Jenkins server.
+    command: java -Djava.awt.headless=true -jar /srv/jenkins/jenkins.war
+    startup: enabled
+parts:
+  add-user:
+    plugin: nil
+    overlay-script: |
+      groupadd -R $CRAFT_OVERLAY --gid 2000 jenkins
+      useradd -R $CRAFT_OVERLAY --system --gid 2000 --uid 2000 --home /srv/jenkins jenkins
+  jenkins:
+    plugin: nil
+    build-packages:
+      - ca-certificates-java
+      - curl
+      - libnss3
+      - unzip
+    overlay-packages:
+      - bash
+      - ca-certificates-java
+      - fonts-dejavu-core
+      - libfontconfig1
+      - openjdk-11-jre-headless
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/{srv/jenkins/,etc/default/jenkins/}
+      chown 2000:2000 ${CRAFT_PART_INSTALL}/srv/jenkins
+      # Use jenkins war rather than apt install for easier Jenkins version control.
+      cd ${CRAFT_PART_INSTALL}/srv/jenkins
+      curl -sLO https://mirrors.jenkins.io/war/2.403/jenkins.war
```

Comment:
We hardcode the version we download here... isn't there a more elegant way ?

Reply:
I could take the variable out into an environment variable for the build step. Would that be a bit more explicit?

Reply:
Part of the intent is actually to hardcode the Jenins version (major.minor). The spec outlines the reason for it if you are interested

Reply:
refactored out to be a build environment variable. thank you!